python/graphstorm/sagemaker/s3_utils.py

```diff
@@ -3,53 +3,101 @@
 """
 import logging
 import os
+from typing import List
 
 import boto3
-import sagemaker
 from sagemaker.s3 import S3Downloader, S3Uploader
+from sagemaker.session import Session
 
 def download_data_from_s3(input_s3, local_data_path, sagemaker_session=None):
     """ Download  data info from S3
 
     Parameters
     ----------
     input_s3: str
-        S3 uri of the input file
+        S3 prefix of the input file(s)
     local_data_path: str
-        Local file path to store the downloaded file.
+        Local file path under which to store the downloaded file(s).
     sagemaker_session: sagemaker.session.Session
         sagemaker_session to run download
     """
     logging.debug("download %s into %s", input_s3, local_data_path)
     if not sagemaker_session:
         boto_session = boto3.Session(region_name=os.environ['AWS_REGION'])
-        sagemaker_session = sagemaker.Session(boto_session)
+        sagemaker_session = Session(boto_session)
     try:
         S3Downloader.download(input_s3,
             local_data_path, sagemaker_session=sagemaker_session)
     except Exception as e: # pylint: disable=broad-except
         raise RuntimeError(f"Error while downloading {input_s3}: {e}")
 
 def upload_file_to_s3(s3_data_path, local_data_path, sagemaker_session=None):
-    """ Upload data info to S3
+    """ Upload local data to S3 prefix
 
     Parameters
     ----------
-    data_path_s3_path: str
-        S3 uri to upload the data
+    s3_data_path: str
+        S3 prefix to upload the data under
     local_data_path: str
-        Path to local data
+        Path to local data that we will upload, filenames will be appended
+        to S3 prefix during upload
     sagemaker_session: sagemaker.session.Session
         sagemaker_session to run upload
     """
     s3_data_path = s3_data_path[:-1] if s3_data_path.endswith('/') else s3_data_path
     logging.debug("upload %s into %s", local_data_path, s3_data_path)
     if not sagemaker_session:
         boto_session = boto3.Session(region_name=os.environ['AWS_REGION'])
-        sagemaker_session = sagemaker.Session(boto_session)
+        sagemaker_session = Session(boto_session)
     try:
         ret = S3Uploader.upload(local_data_path, s3_data_path,
             sagemaker_session=sagemaker_session)
     except Exception as e: # pylint: disable=broad-except
         raise RuntimeError(f"Error uploading data to {s3_data_path}: {e}")
     return ret
+
+def list_s3_files(s3_file_path, sagemaker_session=None) -> List[str]:
```

Comment:
We are no longer using this function but it's a good to have. I would suggest leaving it here for future use.